src/localization/gps_linearization.py

```diff
@@ -30,115 +34,234 @@ class GPSLinearization:
-            self.publish_pose()
+        ref_coord = np.array([self.ref_lat, self.ref_lon, self.ref_alt])
 
-    def imu_callback(self, msg: ImuAndMag):
-        """
-        Callback function that receives IMU messages and publishes the linearized pose.
+        right_cartesian = np.array(
+            geodetic2enu(right_gps_msg.latitude, right_gps_msg.longitude, right_gps_msg.altitude, *ref_coord, deg=True)
+        )
+        left_cartesian = np.array(
+            geodetic2enu(left_gps_msg.latitude, left_gps_msg.longitude, left_gps_msg.altitude, *ref_coord, deg=True)
+        )
 
-        :param msg: The Imu message containing IMU data that was just received
-        """
-        self.last_imu_msg = msg
+        pose = GPSLinearization.compute_gps_pose(right_cartesian=right_cartesian, left_cartesian=left_cartesian)
 
-        if self.last_gps_msg is not None:
-            self.publish_pose()
+        # if imu has not timed out, use imu roll and pitch, calculate offset if fix status is RTK_FIX
+        if (self.imu_has_timeout == False and self.last_imu_orientation is not None):
+            imu_quat = self.last_imu_orientation
+            gps_quat = pose.rotation.quaternion
 
-    @staticmethod
-    def get_linearized_pose_in_world(
-        gps_msg: NavSatFix, imu_msg: ImuAndMag, ref_coord: np.ndarray
-    ) -> Tuple[SE3, np.ndarray]:
-        """
-        Linearizes the GPS geodetic coordinates into ENU cartesian coordinates,
-        then combines them with the IMU orientation into a pose estimate relative
-        to the world frame, with corresponding covariance matrix.
-
-        :param gps_msg: Message containing the rover's GPS coordinates and their corresponding
-                        covariance matrix.
-        :param imu_msg: Message containing the rover's orientation from IMU, with
-                        corresponding covariance matrix.
-        :param ref_coord: numpy array containing the geodetic coordinate which will be the origin
-                          of the tangent plane, [latitude, longitude, altitude]
-        :returns: The pose consisting of linearized GPS and IMU orientation, and the corresponding
-                  covariance matrix as a 6x6 numpy array where each row is [x, y, z, roll, pitch, yaw]
-        """
-        # linearize GPS coordinates into cartesian
-        cartesian = np.array(geodetic2enu(gps_msg.latitude, gps_msg.longitude, gps_msg.altitude, *ref_coord, deg=True))
+            imu_yaw_quat = quaternion_from_euler(0, 0, euler_from_quaternion(imu_quat)[2], "sxyz")
+            gps_yaw_quat = quaternion_from_euler(0, 0, euler_from_quaternion(gps_quat)[2], "sxyz")
+
+            if (right_rtk_fix_msg.RTK_FIX_TYPE == RTKStatus.RTK_FIX and left_rtk_fix_msg.RTK_FIX_TYPE == RTKStatus.RTK_FIX):
+                self.rtk_offset = quaternion_multiply(gps_yaw_quat, quaternion_conjugate(imu_yaw_quat))
+                imu_without_yaw = quaternion_multiply(quaternion_conjugate(imu_yaw_quat), imu_quat)
```

Comment:
I think we can get rid of this and just make the else statement happen no matter what